cirq/google/calibration/workflow.py

```diff
@@ -11,21 +11,210 @@
+        if not isinstance(op, GateOperation):
+            raise IncompatibleMomentError('Moment contains operation different than GateOperation')
+
+        if isinstance(op.gate, MeasurementGate):
+            measurement = True
+        elif isinstance(op.gate, SingleQubitGate):
+            single_qubit = True
+        else:
+            translated_gate = gates_translator(op.gate)
+            if translated_gate is None:
+                raise IncompatibleMomentError(
+                    f'Moment {moment} contains unsupported non-single qubit operation {op}'
+                )
+            elif gate is not None and gate != translated_gate:
+                raise IncompatibleMomentError(
+                    f'Moment {moment} contains operations resolved to two different gates {gate} '
+                    f'and {translated_gate}'
+                )
+            else:
+                gate = translated_gate
+            pair = cast(
+                Tuple[Qid, Qid], tuple(sorted(op.qubits) if pairs_in_canonical_order else op.qubits)
+            )
+            pairs.append(pair)
+
+    if gate is None:
+        # Either empty, single-qubit or measurement moment.
+        return None
+
+    if gate is not None and (measurement or single_qubit):
+        raise IncompatibleMomentError(
+            f'Moment contains mixed two-qubit operations and '
+            f'single-qubit operations or measurement operations.'
+        )
+
+    return FloquetPhasedFSimCalibrationRequest(
+        pairs=tuple(sorted(pairs) if pairs_sorted else pairs), gate=gate, options=options
+    )
+
+
+def floquet_characterization_for_circuit(
+    circuit: Circuit,
+    options: FloquetPhasedFSimCalibrationOptions = FloquetPhasedFSimCalibrationOptions.all_except_for_chi_options(),
+    gates_translator: Callable[[Gate], Optional[FSimGate]] = sqrt_iswap_gates_translator,
+    merge_sub_sets: bool = True,
+    initial: Optional[Tuple[List[FloquetPhasedFSimCalibrationRequest], List[Optional[int]]]] = None,
+) -> Tuple[List[FloquetPhasedFSimCalibrationRequest], List[Optional[int]]]:
+    """Extracts a minimal set of Floquet characterization requests necessary to characterize given
+    circuit.
+
+    The circuit can only be composed of single qubit operations, measurement operations and
+    operations supported by gates_translator.
+
+    Args:
+        circuit: Circuit to characterize.
+        options: Options that are applied to each characterized gate within a moment. Defaults
+            to all_except_for_chi_options which is the broadest currently supported choice.
+        gates_translator: Function that translates a gate to a supported FSimGate which will undergo
+            characterization. Defaults to sqrt_iswap_gates_translator.
+        merge_sub_sets: Whether to merge moments that can be characterized at the same time
+            together.
+        initial: The characterization requests obtained by a previous scan of another circuit. This
```

Comment:
this description is simple but the type signature is complex. Explain how to get one of these?

Reply:
Done.